### DISCUSSION TITLE
Weather

### DISCUSSION THREAD

**Discussion by nodiscardme** [tanujnotes/Olauncher#147] [Ideas]
Hey, I've been using this app for over a month now and it is great so far. However, I wanted one tiny feature from the Default Android Launcher, where it would show weather right above the time widget. Otherwise I have no access to the quick weather update through the home screen. I am sure it would help thousands of other users like me. Waiting for the tweak.

Thank you.

**Comment by DadiBit** [comment]
[Open-Meteo](https://open-meteo.com/) is an interesting API, which does not require any authentication key and could allow great customization over the information that the launcher could show.
I think it shouldn't be too hard to implement the API; there might be some issues with the GPS coordinates, so I would recommend adding an option for a static position (which would also reduce battery usage)...
### EDIT
IMHO, a query every 4 hours should be good enough (if you tap on the weather "widget"/icon you could manually refresh it, long pressing it would open the weather settings). Ideally, the app should cache the API responses: an hourly data cache (temperatures?), and a daily data cache (weather code, for a possible "widget"/icon); this way the user could always see the information, even offline.
In the future, it could also be nice to pick different APIs (eg: [OpenWeather](https://openweathermap.org/api)), but it might just get the launcher too bloated.

**Comment by tanujnotes** [comment]
I'd say implementing a weather API is out of scope for this launcher. People already have their favourite weather apps so the best way to solve this is to add widget support. Currently I don't have any timeline widgets but I do want to add it at some point.

**Reply by Broxzier** [reply]
Great! I just started using this launcher, and being able to place my favourite weather widget would be brill!
